sdcm/send_email.py

```diff
@@ -145,11 +145,29 @@ class LongevityEmailReporter(BaseEmailReporter):
               'username', 'nodes']
 
     def build_report(self, email_data):
-        report_file = os.path.join(self.logdir, 'email_report.html')
-        self.save_html_to_file(email_data, report_file)
         email_data['short_report'] = True
         html = self.render_to_html(email_data)
-        return html, (report_file, )
+        attachments = self.build_report_attachments(email_data)
+        return html, attachments
+
+    def build_report_attachments(self, email_data):
+        report_file = os.path.join(self.logdir, 'email_report.html')
+        self.save_html_to_file(email_data, report_file)
+        attachments = (report_file, )
+        if os.path.getsize(report_file) > 20971520:  # 20Mb
+            if ['test_status'] in email_data and len(email_data['test_status']) > 2 and len(
+                    email_data['test_status'][1]) > 100:
+                #  Reduce number of records in test_status[1] to 100
+                tmp = []
+                for line_number in range(min(100, len(email_data['test_status'][1]))):
+                    tmp.append(email_data['test_status'][1][line_number])
+                tmp.append('List of events has been cut due to the email size limit')
+                email_data['test_status'][1] = tmp
+                self.save_html_to_file(email_data, report_file)
+            if os.path.getsize(report_file) > 20971520:  # 20Mb
+                #  Do not attach any file if it is to large
+                return tuple()
```

Comment:
If the report is still too big, maybe add attachment with that info ? Like in the case cutting to 100